src/parser/config/nodeTypesConfig.json

```diff
@@ -7036,5 +7036,1433 @@
+    },
+    {
+        "expression": "self",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb", "rs"]
+    },
+    {
+        "expression": "string_array",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "subshell",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "symbol_array",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "_pattern_expr",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "_lhs",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "begin",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "case",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "case_match",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "chained_string",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "for",
+        "metrics": ["complexity"],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "hash",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "if",
+        "metrics": ["complexity"],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "method",
+        "metrics": ["complexity", "functions"],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "parenthesized_statements",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "redo",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "retry",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "singleton_class",
+        "metrics": [],
+        "type": "statement",
+        "category": "",
+        "languages": ["rb"]
+    },
+    {
+        "expression": "singleton_method",
+        "metrics": [],
```

Comment:
Are these also relevant for functions metric?

Reply:
I have checked. We should indeed count them. Accordingly, I have added a mapping to functions and complexity as well as test cases.